src/com/aca/raf/management/Manager.java

```diff
@@ -0,0 +1,150 @@
+package com.aca.raf.management;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+public class Manager {
+    private final List<Gadget> gadgets;
+    private List<Gadget> notWorkingItems;
+    private int soldAmount = 0;
+
+    public Manager(List<Gadget> gadgets) {
+        this.gadgets = gadgets;
+    }
+
+    public List<Gadget> getNotWorkingItems() {
+        return notWorkingItems;
+    }
+
+    public void SellItem(Class classOfGadget){
+        for (Gadget gadget: gadgets){
+            if (gadget.getClass() == classOfGadget){
+                gadget.setCount(gadget.getCount() - 1);
+                soldAmount += gadget.getPrice();
+            }
+        }
+    }
+
+    public List<VideoRecorder> getAllVideoRecorders(){
+        List<VideoRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget: gadgets){
+            if (gadget instanceof VideoRecorder){
+                recorders.add((VideoRecorder) gadget);
+            }
+        }
+        return recorders;
+    }
+
+    public List<VideoRecorder> getAllVideoRecordersByType(VideoRecorder.VideoQuality videoQuality) {
+        List<VideoRecorder> recorders = new ArrayList<>();
+        for (Gadget gadget: gadgets){
+            if (gadget instanceof VideoRecorder){
+                VideoRecorder videoRecorder = (VideoRecorder) gadget;
+                if (videoRecorder.getVideoQuality() == videoQuality){
+                    recorders.add(videoRecorder);
+                }
+            }
+        }
+        return recorders;
+    }
+
+    public int getSoldAmount() {
+        return soldAmount;
+    }
+
+    public int getInStockAmount(){
+        int amount = 0;
+        for (Gadget gadget: gadgets){
+            amount += gadget.getCount() * gadget.getPrice();
+        }
+        return amount;
+    }
+
+    public void checkGadgets(){
+        notWorkingItems = new ArrayList<>();
+        Random rand = new Random();
+        for (Gadget gadget: gadgets) {
+            int randInt = rand.nextInt(100);
+            if (gadget instanceof VideoRecorder) {
+                if (randInt % 2 == 0){
+                    printItemsOfVideoRecordingGadget(gadget);
+                }else {
+                    notWorkingItems.add(gadget);
+                }
+            }else if (gadget instanceof AudioRecorder){
+                if (randInt % 2 == 0){
+                    printItemsOfAudioRecordingGadget(gadget);
```

Comment:
why do you have these print... functions if you could implement that logic into toString for each type